### ISSUE TITLE
Shared libraries?

### ISSUE THREAD

**Issue by limburgher** [erincatto/box2d#621]
Hi! I maintain the Fedora package for Box2D. I'm looking at upgrading from 2.3.1 to 2.4.0, but I can't get the new cmake build system to build shared libraries, only static. Is this by design, or have I missed something? Thank you!

**Comment by erincatto** [comment]
This is in the FAQ: https://box2d.org/documentation/md__d_1__git_hub_box2d_docs__f_a_q.html

Did you get shared libraries working for 2.3.1? If so, maybe 2.4.0 would work as shared library. I mainly develop on Windows and I know it would not work as a DLL. I do take PRs for cmake changes if you get it building as a shared library. This option should not be available on Windows.

**Comment by limburgher** [comment]
We did, 2.2.1 as well, with -DBOX2D_INSTALL=ON -DBOX2D_BUILD_SHARED=ON.

I'm not expert enough in cmake to easily see what would be needed to adapt to 2.3.1.

**Comment by erincatto** [comment]
Duplicate of #622

**Comment by erincatto** [comment]
2.3.1 had cmake installation scripts that I did not write.

**Comment by limburgher** [comment]
Do you know who did?

**Comment by erincatto** [comment]
Here is one of the commits:
https://github.com/erincatto/box2d/commit/f2aadfeb0ec7483efbe8ec214ebba65f0abcd36d

**Comment by limburgher** [comment]
Hmm. Way past my cmake expertise.

**Comment by jube** [comment]
I can make a PR for this issue, if you want.

**Comment by limburgher** [comment]
Please do! I can then adopt it even if Erin would rather not merge it for some reason.

**Comment by svenstaro** [comment]
I'm the Arch Linux package maintainer and I too would appreciate shared library support.

**Comment by jube** [comment]
@svenstaro if you want to test the patch I provided in #630 and #631, you can use the https://github.com/jube/box2d/tree/install_and_shared branch and see if it works for you

**Comment by svenstaro** [comment]
Compilation works ok but it doesn't set the correct sonames. I'd expect a `libbox2d.so.2` and `libbox2d.so.2.4` to be installed but they aren't. Also docs aren't installed for me. Perhaps it would be cool to also install the examples and testbed?

Apart from that it seems to be ok.

**Comment by jube** [comment]
My bad, you are totally right. I forgot to add the necessary properties. I made the corrections and pushed everything (in the two PR and in the special branch). I installed the documentation but not the examples (there isn't any) and tesbed (as it would require installing many dependencies which are embedded in the repository).

I don't think it's a good idea to install the testbed. I think it's a tool for Erin to test the library. I don't know if it's meant to be public.
